### PR TITLE
[FIX] im_livechat: use current partner in livechat

### PR DESCRIPTION
**Pull request by phenix-factory**
Current code can't work since session doesn't contain partner_id.

task-3010650

### CODE REVIEW

**Comment by bouvyd** [comment]
@robodoo r-

this is incompatible with commit 2a5c6c4bc494faecccb86e4b8ad33dab3476b363 which re-introduced the `session` variable at line 78, thus whilst this PR does not conflict, it deletes a variable used in the code; needs a rebase and to handle the removal of the session again. You could not have seen it, since this PR and the conflicting commit where never on the same branch before the current staging - but let's avoid having robodoo working all night on a split :)

cc @ThanhDodeurOdoo

**Comment by phenix-factory** [comment]
@bouvyd I rebased this PR, shoudl be good now.

File: addons/im_livechat/static/src/legacy/models/public_livechat_message.js [hunk 1]
```diff
@@ -260,7 +261,7 @@ const PublicLivechatMessage = Class.extend({
      * @return {boolean}
      */
     _isMyselfAuthor() {
-        return this.hasAuthor() && (this.getAuthorID() === session.partner_id);
+        return this.hasAuthor() && (this.getAuthorID() === this.currentPartnerId);
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/legacy/models/public_livechat_message.js:264]: Just compare with `this.messaging.publicLivechatGlobal.options.current_partner_id` (don't define `this.currentPartnerId`)

**Review by alexkuhn — CHANGES_REQUESTED** [review]

File: addons/im_livechat/static/src/legacy/models/public_livechat_message.js (hunk 1 above)
Comment (phenix-factory) [reply addons/im_livechat/static/src/legacy/models/public_livechat_message.js]: Done

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**